android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java

```diff
@@ -400,6 +381,9 @@ public void onConfigurationChanged(Configuration newConfig) {
 
     int bannerHeight = KMManager.getBannerHeight(context);
     int oskHeight = KMManager.getKeyboardHeight(context);
+    if (this.htmlBannerString != null && !this.htmlBannerString.isEmpty()) {
+      loadJavascript(KMString.format("setBannerHTML('%s')", this.htmlBannerString));
```

Comment:
Will this work with arbitrary HTML, even HTML that contains `'` characters? i.e. are they escaped?

Reply:
And shouldn't this read:

```suggestion
      setHTMLBanner(this.htmlBannerString);
```

Reply:
This is setting up a WebView operation.  Don't forget about that JS queue we set up a while back; this adds an entry to that queue (when necessary).  We want the equivalent of what you've typed when within the WebView; the trick is making sure the values marshal across the WebView boundary properly in order to do so.

We'd probably be best off by using `JSON.stringify` encoding directly on `this.htmlBannerString` here, then `JSON.parse()` within the inner `setBannerHTML` function from the WebView-internal receiver.

Reply:
Not sure if you are responding to my suggestion here @jahorton?

> This is setting up a WebView operation. Don't forget about that JS queue we set up a while back; this adds an entry to that queue (when necessary). 

See ll.387-389. those are also all calling `loadJavascript` too

> (We _are_ on the Java / Android side here; your suggestion would be a JS-side call.)

See ll.650-654.

Reply:
FWIW, all of these should be setup as a single `loadJavascript()` call, appending each of the functions, so we are not doing multiple callbacks on keyboard load. That should improve performance and probably reduce flicker too.

Reply:
See #8534 - what has been noted here is a very likely contributor.  If those calls are kept separately, each call is a _separate async call_ into the WebView, and those delays could **very easily** contribute to layout reflow / flashing... especially since each affects the keyboard's layout.

Reply:
For proper escaping, here's the basic pattern:

https://github.com/keymanapp/keyman/blob/5a0c1a38a09047ff7e45a4bfc0b3b35dcd5b1378/android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java#L618-L636

Something like:

```suggestion
      loadJavascript(KMString.format("setBannerHTML(%s)",
        JSONObject.quote(this.htmlBannerString)));
```

Reply:
> FWIW, all of these should be setup as a single loadJavascript() call, appending each of the functions, so we are not doing multiple callbacks on keyboard load. That should improve performance and probably reduce flicker too. (That would undo my suggestion of calling setBannerHTML but I would be okay with that)

This will be addressed on separate PR